File: javascript/005/main.js
/*
 * ProjectEuler 005
 * JavaScript / main.js
 * copyright (c) 2015 Susisu
 *
 * Node.js v0.12.0
 * `node main.js`
 */

function main() {
    var product = 1;
    for (var n = 1; n <= 20; n++) {
        product *= n / gcd(product, n);
    }
    console.log(product);
}

function gcd(m, n) {
    if (m < n) {
        return gcd(n, m);
    }
    else if (n === 0) {
        return m;
    }
    else {
        return gcd(n, m % n);
    }
}

main();

File: lua/005/main.lua
--[[
    ProjectEuler 005
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local p = 1
    for i = 1, 20 do
        p = p / gcd(p, i) * i
    end
    print(p)
end

function gcd(a, b)
    if a < b then
        return gcd(b, a)
    elseif a % b == 0 then
        return b
    else
        return gcd(b, a % b)
    end
end

main()
